postgres/src/main/java/org/jdbi/v3/postgres/InetColumnMapper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import org.jdbi.v3.core.mapper.ColumnMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.postgresql.util.PGobject;
+
+/**
+ * A column mapper which maps Postgres's {@code inet} type to Java's {@link InetAddress}.
+ */
+class InetColumnMapper implements ColumnMapper<InetAddress> {
+    @Override
+    public InetAddress map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
+        String val = Optional.ofNullable(r.getObject(columnNumber))
+            .map(PGobject.class::cast)
+            .map(PGobject::getValue).orElse(null);
+
+        try {
+            return val == null ? null : InetAddress.getByName(val);
+        } catch (UnknownHostException ex) {
+            throw new IllegalArgumentException(String.format(
+                "could not get %s from %s (%s)", InetAddress.class.getSimpleName(), val, ex));
```

Comment:
use the (message, throwable) constructor and not format the exception as part of the message. This will break the exception chain

also, in the InternetMapperFactory, we throw a MappingException, so this should match

Reply:
`InetColumnMapper` ist gone

I find quite a few of mappers that throw `IllegalArgumentException`. Those should be reviewed then and possibly replaced by `MappingException`. I'll take a note and may provide a separate pull request.